profiler/data_collector.rst

```diff
@@ -73,6 +78,12 @@ The getters are added to give the template access to the collected information.
 
 .. caution::
 
+    If the data that is not directly related to the request or response, you need to make the data accessible to your DataCollector.
+    This can be achieved by injecting the service into your DataCollector.
```

Comment:
I think we can be a bit more specific what we mean with "the service" (something like "by injecting a service that provides this information into your data collector).